### DISCUSSION TITLE
xray-junit-extensions (Not injecting changes to Test Issue Type)

### DISCUSSION THREAD

**Discussion by shrubom** [Xray-App/xray-junit-extensions#45] [Show and tell]
Hi, 
I need help in how to exactly use xray-junit-extension.
The dependency I am using is  `        <dependency>
            <groupId>app.getxray</groupId>
            <artifactId>xray-junit-extensions</artifactId>
            <version>0.7.1</version>
            <scope>test</scope>
        </dependency>`
       
Test Method:
   
   @Test
    @DisplayName("Error Message")
    @Description("Validating error message")
    @XrayTest(key = "TEST-336", summary = " Validating error message", description = " Validating error message")
    @Requirement("TEST-280")
    @Tags({@Tag("Regression"), @Tag("TEST-336")})
    public void validateErrMessage(XrayTestReporter xrayReporter) throws InterruptedException, IOException {
        xrayReporter.addComment("This test Validates error message for Foreign Country for Lithuania/Estonia/Latvia");
        //Test Code
    }
    
Jenkins: 
![image](https://github.com/Xray-App/xray-junit-extensions/assets/44477101/228357e8-6268-4b21-9fd4-32f454a48ab5)

Console o/p: 
![image](https://github.com/Xray-App/xray-junit-extensions/assets/44477101/32963c3e-a0ef-43ae-9047-fe5b1228650a)

As you can see it is not updating the Issue Key - TEST-336 But creating a new Test Issue Key - TEST-344

Can someone please advice if the implementation is correct ? I am not sure what I am missing here.

**Comment by bitcoder** [comment]
I think you're uploading the standard surefire JUnit reports and not the enhanced JUnit XML report generated by the listener on this project.
Can you please share:

1. the JUnit XML report that was generated?
2. where exactly are you generating the test report to? can you share your src/test/resources/xray-junit-extensions.properties file? By default the report won't be under the surefire directory but under `target/reports/TEST-junit-jupiter.xml` ;

**Reply by shrubom** [reply]
Hi, 

Yes. I am not using any enhanced JUnit XML report.

1. I am assuming the JUnit XML generated is somewhat like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://maven.apache.org/surefire/maven-surefire-plugin/xsd/surefire-test-report-3.0.xsd" version="3.0" name="tests.CustomerDocumentTest" time="44.753" tests="1" errors="0" skipped="0" failures="0">
<properties>--------<properties>
  
  <testcase name="validateErrMessageForForeignCountry(XrayTestReporter)" classname="tests.CustomerDocumentTest" time="40.046"/>
</testsuite>
```

**I got it from this location:** target/surefire-reports/TEST-tests.CustomerDocumentTest.xml

2. Unfortunately I dont see any folder named "reports" under the target folder. 

3. xray-junit-extensions.properties:

report_filename=TEST-tests.CustomerDocumentTest
report_directory=target
add_timestamp_to_report_filename=true
report_only_annotated_tests=true

4. junit-platform.properties:

junit.jupiter.extensions.autodetection.enabled=true
junit.jupiter.extensions.autodetection.discovery-mode = annotated
junit.jupiter.execution.parallel.enabled = ${PARALLEL}
junit.jupiter.execution.parallel.mode.default = concurrent
junit.jupiter.execution.parallel.mode.classes.default = concurrent
junit.jupiter.execution.parallel.config.strategy = fixed
junit.jupiter.execution.parallel.config.fixed.parallelism = ${FORK_PROCESSES}

**Comment by bitcoder** [comment]
It seems you're not using the report generated by this plugin; to to that you need to enable the listener EnhancedLegacyXmlReportGeneratingListener as mentioned [here](https://github.com/Xray-App/xray-junit-extensions?tab=readme-ov-file#how-to-use).

Example: see [this tutorial](https://github.com/Xray-App/tutorial-java-junit5-selenium/blob/main/src/test/resources/META-INF/services/org.junit.platform.launcher.TestExecutionListener)

Note: You can always clone the tutorials at https://github.com/Xray-App/xray-junit-extensions and then adjust it to your needs.  If you got it working, you'll have a new JUnit XML report where its contents will have those attributes like the test_key as `<property>`  element.

**Reply by shrubom** [reply]
Thank you so much. I will try to clone this and see if it works. If not I will contact you again. :)

**Comment by shrubom** [comment]
Hi,
Update on the discussion.
 
Finally got it working. :)

1. Now I can add comments to tests
2. Update the summary of the test 
3. Add Description to the test 
4. Add labels to the test 

So i manually added the file(org.junit.platform.launcher.TestExecutionListener) in the location -> src/test/resources/META-INF/services/org.junit.platform.launcher.TestExecutionListener

This generated the JUnit xml  -> target/TEST-tests.CustomerDocumentTest.xml

**Comment by shrubom** [comment]
Thank you so much :)
